service/src/main/kotlin/xyz/lebkuchenfm/external/discord/DiscordClient.kt

```diff
@@ -53,13 +59,41 @@ class DiscordClient(config: ApplicationConfig, private val commandExecutorServic
             .map { it.message }
             .filter { it.channelId.value.toString() == channelId }
             .filter { it.content.startsWith(commandPrompt) }
+            .filter { it.author != null }
             .filter { it.author?.isBot == false }
             .onEach {
-                // TODO: Get user by Discord ID.
-                val context = ExecutionContext(UserSession(name = "FAKE USER TODO"))
-                val result = commandExecutorService.executeFromText(it.content, context)
-                it.reply {
-                    content = result.message.markdown
+                val author = requireNotNull(it.author)
+                val user = userService.getByDiscordId(author.id.toString())
+
+                // TODO: Add support for login command.
+                val isLoginCommand = false
+
+                when {
+                    isLoginCommand -> {
+                        val context = ExecutionContext(UserSession(name = "unknown"))
+                        val result = commandExecutorService.executeFromText(it.content, context)
+                        it.reply {
+                            content = result.message.markdown
+                        }
+                    }
+
+                    user != null -> {
+                        val context = ExecutionContext(UserSession(user.data.name))
+                        val result = commandExecutorService.executeFromText(it.content, context)
+                        it.reply {
+                            content = result.message.markdown
+                        }
+                    }
+
+                    else -> {
+                        // TODO: Add support for login command.
+                        it.reply {
+                            content = """
+                                You have to connect your Discord account with LebkuchenFM.
+                                Use `$commandPrompt TODO LOGIN_COMMAND_KEY <lebkuchen-fm-username>` to login.
```

Comment:
shouldn't we just pass command to execution service and allow it to response with help message? 🤔

Reply:
I don't think so. This specific message is a 403 of sorts for Discord. Your user is not assigned to any LebkuchenFM user, so you're basically not logged in.